src/main/scala/firrtl/passes/UpdateDuplicateMemMacros.scala

```diff
@@ -33,35 +83,41 @@ object MemTransformUtils {
     updateMemPortMap(m.readers, rFields, "R")
     updateMemPortMap(m.writers, wFields, "W")
     updateMemPortMap(m.readwriters, rwFields, "RW")
-    memPortMap.toMap
+    memPortMap
   }
+
   def createMemProto(m: DefMemory) = {
     val rports = (0 until m.readers.length) map (i => s"R$i")
     val wports = (0 until m.writers.length) map (i => s"W$i")
     val rwports = (0 until m.readwriters.length) map (i => s"RW$i")
-    m.copy(readers = rports, writers = wports, readwriters = rwports)
+    m copy (readers = rports, writers = wports, readwriters = rwports)
   }
 
-  def updateStmtRefs(s: Statement, repl: Map[String, Expression]): Statement = {
-    def updateRef(e: Expression): Expression = e map updateRef match { 
-      case e => repl getOrElse (e.serialize, e) 
+  def updateStmtRefs(repl: MemPortMap)(s: Statement): Statement = {
+    def updateRef(e: Expression): Expression = {
+      val ex = e map updateRef
+      repl getOrElse (ex.serialize, ex)
     }
+
     def hasEmptyExpr(stmt: Statement): Boolean = {
       var foundEmpty = false
       def testEmptyExpr(e: Expression): Expression = {
-        e map testEmptyExpr match {
+        e match {
           case EmptyExpression => foundEmpty = true
           case _ =>
         }
-        e // map must return; no foreach
+        e map testEmptyExpr // map must return; no foreach
       }
       stmt map testEmptyExpr
```

Comment:
Just want to be sure, and I'm sure @jackkoenig already mentioned this to me. stmt map testEmptyExpr goes through all of the expressions in the stmt? So a map on a statement could be "sub statements" i.e. statements in Block or it can be Expressions?